test/functional/test_framework/address.py

```diff
@@ -176,3 +189,18 @@ def check_base58(data, version):
         check_base58(bytes.fromhex('0041c1eaf111802559bad61b60d62b1f897c63928a'), 0)
         check_base58(bytes.fromhex('000041c1eaf111802559bad61b60d62b1f897c63928a'), 0)
         check_base58(bytes.fromhex('00000041c1eaf111802559bad61b60d62b1f897c63928a'), 0)
+
+
+    def test_bech32_decode(self):
+        def check_bech32_decode(payload, version):
+            hrp = "tb"
```

Comment:
Instead of hardcoding the hrp here, why not have `check_bech32_decode()` take `hrp` as a parameter, then we can add `bc1` test vectors instead of being limited to `tb1`?